src/chat/chatter.py

```diff
@@ -183,15 +185,25 @@ def update(self):
         # Status icon handling
         url = client.instance.urls.get(player.login)
         if url:
+            icon_url = None
             if url.scheme() == "fafgame":
-                self.statusItem.setIcon(util.icon("chat/status/lobby.png"))
+                icon_url = "chat/status/lobby.png"
                 self.statusItem.setToolTip("In Game Lobby<br/>"+url.toString())
             elif url.scheme() == "faflive":
-                self.statusItem.setIcon(util.icon("chat/status/playing.png"))
+                icon_url = "chat/status/playing.png"
                 self.statusItem.setToolTip("Playing Game<br/>"+url.toString())
-        else:
-            self.statusItem.setIcon(QtGui.QIcon())
-            self.statusItem.setToolTip("Idle")
+
+            if icon_url:
+                mapname = url.queryItemValue('map')
+                icon = maps.preview(mapname)
+                if not icon:
+                    self.lobby.client.downloader.downloadMap(mapname, self)
```

Comment:
Won't this download all maps being played by anyone online at a given time?

Reply:
Yup.

For this feature to work sanely, you're going to want some way to get map icons without getting the map itself. The new API probably does that?

Reply:
Not a clue. First thing that jumps to mind as a sane, if not comprehensive or complete, way to do it would be to include a map icons download package with the client installer by default, including most of the common ones, and just use the swords if you don't have them? 

API route to grab by mapname would be better of course.

Reply:
> API route to grab by mapname would be better of course.

Really, this is the only sane approach. An API route to serve up the icon. For bonus points, you could request it at a certain size and have the server cache requested images.

>  include a map icons download package with the client installer by default, including most of the common ones, and just use the swords if you don't have them?

You could. But adding an API route - if it doesn't already exist - should be very straightforward. I'd suggest going right for the final solution.

Reply:
AFAIK the "downloadMap" method actually downloads the map preview, confusing as it is. The preview images are simply served from content.faforever.com. If you want to add some fancy API routes, then go ahead :)

Reply:
Having something named "downloadMap" only grab the preview image is _fucking stupid_. If true, that needs renaming pronto.